src/writers/s3.sink.hh

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include "sink.hh"
+#include "platform.h"
+#include "common/s3.connection.hh"
+
+#include <miniocpp/types.h>
+
+#include <array>
+#include <string>
+
+namespace acquire::sink::zarr {
+class S3Sink final : public Sink
+{
+  public:
+    S3Sink() = delete;
+    S3Sink(std::string_view bucket_name,
+           std::string_view object_key,
+           std::shared_ptr<common::S3ConnectionPool> connection_pool);
+    ~S3Sink() override;
+
+    bool write(size_t offset,
+               const uint8_t* data,
+               size_t bytes_of_data) override;
+
+  private:
+    std::string bucket_name_;
+    std::string object_key_;
+
+    std::shared_ptr<common::S3ConnectionPool> connection_pool_;
+
+    // multipart upload
+    std::array<uint8_t, 5 << 20> part_buffer_;
```

Comment:
I would restore the comment or create a variable (`static constexpr size_t five_mb_size = 5 << 20`)  so it's obvious what this number represents.